src/dmd/link.d

```diff
@@ -981,14 +981,15 @@ public int runProgram()
         childpid = fork();
         if (childpid == 0)
         {
-            const(char)* fn = argv[0];
+            const(char)[] fn = argv[0].toDString();
+            // Make it "./fn" if needed
             if (!FileName.absolute(fn))
-            {
-                // Make it "./fn"
                 fn = FileName.combine(".", fn);
-            }
-            execv(fn, argv.tdata());
-            perror(fn); // failed to execute
+            fn.toCStringThen!((fnp) {
+                    execv(fnp.ptr, argv.tdata());
+                    // If execv returns, it failed to execute
+                    perror(fnp.ptr);
+                });
```

Comment:
I don't see a point. There are more lines of code, it is more complex, a lambda is thrown in, and it converts a C string to a D string and then back to a C string.